### ISSUE TITLE
VMR stage2 failed to find a stable package System.CommandLine 

### ISSUE THREAD

**Issue by Winniexu01** [dotnet/source-build#4487]
[Failing build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2485866&view=logs&j=9f4c4083-9864-5575-dedd-fb296084d8e2&t=15f0c853-f021-5549-d93c-f01428803354) (internal Microsoft link)

`Prep the Build` failed:

    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : warning NU1604: Project dependency Microsoft.Extensions.FileSystemGlobbing does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.
    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : warning NU1604: Project dependency Microsoft.Extensions.Logging does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.
    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : warning NU1604: Project dependency Microsoft.Extensions.Logging.Console does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.
    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : warning NU1604: Project dependency System.CommandLine does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results.
    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : error NU1103: Unable to find a stable package System.CommandLine with version 
    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : error NU1103:   - Found 1 version(s) in /tmp/tmp.UFv6tMGnqh [ Nearest version: 2.0.0-beta4.24326.1 ]
    /vmr/eng/tools/BinaryToolKit/BinaryToolKit.csproj : error NU1103:   - Found 0 version(s) in /vmr/.dotnet/library-packs
    
    The build failed. Fix the build errors and run again.

**Comment by ViktorHofer** [comment]
@ellahathaway can you please take a look?

**Comment by tmds** [comment]
I'm seeing the following build failures in our CI on stage2. @ViktorHofer @ellahathaway I assume the root cause is the same?

```
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.csproj : error NU1604: Warning As Error: Project dependency Newtonsoft.Json does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Abstractions does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/Microsoft.TemplateEngine.Orchestrator.RunnableProjects.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Utils does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Abstractions/Microsoft.TemplateEngine.Abstractions.csproj : error NU1604: Warning As Error: Project dependency Microsoft.Extensions.Logging.Abstractions does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateSearch.Common/Microsoft.TemplateSearch.Common.csproj : error NU1604: Warning As Error: Project dependency Newtonsoft.Json does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateSearch.Common/Microsoft.TemplateSearch.Common.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Abstractions does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Utils/Microsoft.TemplateEngine.Utils.csproj : error NU1604: Warning As Error: Project dependency Microsoft.Extensions.Logging.Abstractions does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Utils/Microsoft.TemplateEngine.Utils.csproj : error NU1604: Warning As Error: Project dependency Newtonsoft.Json does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Core/Microsoft.TemplateEngine.Core.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Abstractions does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Core/Microsoft.TemplateEngine.Core.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Utils does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Core/Microsoft.TemplateEngine.Core.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Utils does not provide an inclusive lower bound for dependency Newtonsoft.Json. An approximate best match of Newtonsoft.Json 13.0.3 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Core.Contracts/Microsoft.TemplateEngine.Core.Contracts.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Abstractions does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj : error NU1604: Warning As Error: Project dependency Microsoft.Extensions.Logging.Abstractions does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj : error NU1604: Warning As Error: Project dependency Newtonsoft.Json does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj : error NU1604: Warning As Error: Project dependency NuGet.Configuration does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj : error NU1604: Warning As Error: Project dependency NuGet.Credentials does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.Edge/Microsoft.TemplateEngine.Edge.csproj : error NU1604: Warning As Error: Project dependency NuGet.Protocol does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Abstractions does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Edge does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Utils does not provide an inclusive lower bound for dependency Microsoft.Extensions.Logging.Abstractions. An approximate best match of Microsoft.Extensions.Logging.Abstractions 2.1.1 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Edge does not provide an inclusive lower bound for dependency Newtonsoft.Json. An approximate best match of Newtonsoft.Json 13.0.3 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Orchestrator.RunnableProjects does not provide an inclusive lower bound for dependency Newtonsoft.Json. An approximate best match of Newtonsoft.Json 13.0.3 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Utils does not provide an inclusive lower bound for dependency Newtonsoft.Json. An approximate best match of Newtonsoft.Json 13.0.3 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Edge does not provide an inclusive lower bound for dependency NuGet.Configuration. An approximate best match of NuGet.Configuration 6.2.4 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Edge does not provide an inclusive lower bound for dependency NuGet.Credentials. An approximate best match of NuGet.Credentials 6.2.4 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
    /home/tester/dotnet/src/templating/src/Microsoft.TemplateEngine.IDE/Microsoft.TemplateEngine.IDE.csproj : error NU1602: Warning As Error: Microsoft.TemplateEngine.Edge does not provide an inclusive lower bound for dependency NuGet.Protocol. An approximate best match of NuGet.Protocol 6.2.4 was resolved. [/home/tester/dotnet/src/templating/Microsoft.TemplateEngine.sln]
```

**Comment by ViktorHofer** [comment]
Yes, this will be fixed with https://github.com/dotnet/sdk/pull/41895. This is now needed because NuGet reverted a change that we asserted on.

**Comment by ellahathaway** [comment]
> Yes, this will be fixed with [dotnet/sdk#41895](https://github.com/dotnet/sdk/pull/41895). This is now needed because NuGet reverted a change that we asserted on.

I noticed that the source-build VMR Stage 2 builds are still failing with the templating errors that @tmds described despite the changes from https://github.com/dotnet/sdk/pull/41895 having flowed in.

[Link to failing build](https://dev.azure.com/dnceng/internal/_build/results?buildId=2488040&view=logs&j=3121c41d-6088-53a5-4ef6-9a86f3460fe9&t=334539f4-dcef-5cad-a240-02997b3146e7&l=1510) (internal Microsoft link)

cc @ViktorHofer
